### ISSUE TITLE
The optimizer uses argument `set_to_none=False` by default

### ISSUE THREAD

**Issue by yongchanghao** [huggingface/accelerate#2471]
The default behavior of the optimizer wrapper takes `set_to_none=False` as shown here:
https://github.com/huggingface/accelerate/blob/97d2168e5953fe7373a06c69c02c5a00a84d5344/src/accelerate/optimizer.py#L117

This may have two issues:
1.  It contradicts the default behavior of PyTorch
2.  It produces more memory footprint as every forward-backward would result in a gradient accumulation

Is there any consideration here to have this design?

**Comment by BenjaminBossan** [comment]
Thanks for bringing this up. I agree that there an argument to be made to use the same default as PyTorch. It appears that the switch in PyTorch was made ~1 year ago in [this PR](https://github.com/pytorch/pytorch/pull/92731) but accelerate hasn't updated the default since then.

The only minor concern I see is that this could theoretically break backwards compatibility, although I don't know if there is any practical concern. Curious what others think.

**Comment by yongchanghao** [comment]
Thanks for the reference. Let me know whether I can help if the decision is to align with PyTorch.
